### PR TITLE
OCPBUGS-34119: Upgrade controller-tools and library-go

### PR DESCRIPTION
**Pull request by honza**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,147 +1,169 @@
 module github.com/openshift/cluster-baremetal-operator
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.3
```
Comment (dtantsur) [inline go.mod:5]: This needs to be dropped

Comment (honza) [reply go.mod]: Done

**Pull request merged**